app/src/main/java/nick/mirosh/newsapp/ui/feed/MainScreenContent.kt

```diff
@@ -74,8 +70,13 @@ fun MainScreenContent(
                     onLikeClick = viewModel::onLikeClick,
                     onSavedArticlesClicked = onSavedArticlesClicked
                 )
+
+            else -> {}
         }
     }
+    AnimatedVisibility(visible = uiState == FeedUIState.Loading, enter = fadeIn(), exit = fadeOut()) {
+        SmileyProgressAnimation()
+    }
 }
```

Comment:
Moved `SmileyProgressAnimation` out of `when` block to make sure `AnimatedVisibility` still runs when state is changed to `!Loading`. Keeping it inside `when` won't trigger the `fadeOut()` animation.